test/product_patterns_test.rb

```diff
@@ -84,6 +84,44 @@
 
       expect(subject.names.sort).to eq([default_pattern1, default_pattern2].sort)
     end
+
+    context "repository parameter has been set" do
+      # get the default patterns only from the repository with id 2
+      subject { Yast::ProductPatterns.new(src: 2) }
+
+      it "returns the default patterns only from the selected repository" do
+        default_pattern1 = "def_pattern1"
```

Comment:
I prefer to use let for this kind of `initializations`, so it becomes clear what's the test setup and what's the test itself. But it's just a matter of taste, I'm fine with that.